Applications/Spire/Include/Spire/Ui/FlatButton.hpp

```diff
@@ -23,8 +23,8 @@ namespace Spire {
         QColor m_text_color;
       };
 
-      //! Signals that the button was clicked.
-      using ClickedSignal = Signal<void ()>;
+      //! Signals that the button was pressed.
+      using PressedSignal = Signal<void ()>;
```

Comment:
I totally agree with you on the point that FlatButton should be derived from QAbstractButton or QPushButton. The FlatButton can be further simplified by subclassing from QAbstractButton or QPushButton, but I don’t think the released() signal has a clearer meaning than pressed or clicked. Besides, could you provide a case where Spire::Signal type might cause some issues.

Reply:
You are welcome to inherit from an existing Qt class. Use ```clicked``` instead of ```pressed``` or ```released```.

Reply:
The implementation signals don't have to match the specification signal names exactly. The most important things are that the implementation behaves correctly according to the UX and UI specs, and that the implementation is clean and maintainable. If there is existing functionality that does what we need, then we can save ourselves the trouble of maintaining our own implementation by using the existing solution.

This brings up a bug in my IconButton implementation, because I used the released signal which is subtly different from the clicked signal.

Reply:
When I replaced Spire::Signal with the clicked() signal, I met a new problem. The function connect_clicked_signal in the original FlatButton class returns a boost connection, and if I change it to QObject::connect() which returns Qt connection, the Qt connection cannot be accepted by the Beam::SignalHandling::ConnectionGroup which contains boost signals and is used in the ColorSelectorDropDown class. It means changing Spire::Signal in the FlatButton class can cause a major change in the ColorSelectorDropDown class.

Reply:
That's fine, that ConnectionGroup is only there to manage the color button connections, so it can be removed. Because the Qt connections are automatically deleted when the object is deleted, we no longer have to manage them. QObject::connect(...) returns a Connection in case we wanted to selectively disconnect the signal, which we don't need here so it can be ignored.

The calls to connect_pressed_signal(...) can be replaced with a call to QObject::connect(...), which will call a lambda that captures the button's color and passes it to on_color_button_clicked(...).

Reply:
I see.